WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -89,10 +90,16 @@ private void handleIntent(Intent intent) {
             actionBar.setDisplayHomeAsUpEnabled(true);
         }
 
+        PostModel targetPost = null;
+        int targetPostId = intent.getIntExtra(EXTRA_TARGET_POST_ID, 0);
+        if (targetPostId > 0) {
+            targetPost = mPostStore.getPostByLocalPostId(intent.getIntExtra(EXTRA_TARGET_POST_ID, 0));
+        }
+
         mPostList = (PostsListFragment) getFragmentManager().findFragmentByTag(PostsListFragment.TAG);
-        if (mPostList == null || siteHasChanged || pageHasChanged) {
+        if (mPostList == null || siteHasChanged || pageHasChanged || targetPost != null) {
             PostsListFragment oldFragment = mPostList;
-            mPostList = PostsListFragment.newInstance(mSite, mIsPage);
+            mPostList = PostsListFragment.newInstance(mSite, mIsPage, targetPost);
```

Comment:
In this case we're passing the `PostModel` directly to a method, it's not being parceled, so https://github.com/wordpress-mobile/WordPress-Android/pull/5856 shouldn't apply.

I decided to use the model for two reasons:
1. We won't create a new instance of the `PostsListFragment` if the post has since been deleted (since it would have a non-0 id, but the `PostModel` obtained from the store would be `null`)
2. This avoids `PostsListFragment.newInstance()` taking an int id argument, which is error-prone and the main case for using models over ids